### PR TITLE
Revamp the README

### LINKED ISSUE #554: Create a docker image on docker hub
HI,

It's possible to add a docker image on docker hub with voxpupuli account ?

The older images was puppetlabs (an archived project https://github.com/puppetlabs/puppet-in-docker)

Regards,

### LINKED ISSUE #579: Link in Readme to Docker package is bad
This link to maintained docker image 

https://github.com/voxpupuli/puppetboard#packages-1

Just goes to a locked down docker hub account. I'm guessing that dude has punted on maintaining it.

### PR DESCRIPTION
**Pull request by gdubicki**
* Add Table of Contents
* Update the main screenshot to show app in v. 2.2.0,
* Remove info about the outdated packages, Docker image, contact methods,
* Move the too long and mostly outdated anyway production setup docs
  into a separate file,
* Shorten and deduplicate,
* Unify the method of header formatting,
* Break long lines.

\+

* Just point to camptocamp's Docker image - it's maintained and the provider is trusted,

Fixes #554, #574 (sort of) and #579.

### CODE REVIEW

**Comment by gdubicki** [comment]
Instead of reading through the diff of this PR I suggest going through the new README using the Github markdown viewer, here: https://github.com/gdubicki/puppetboard/tree/update_readme .

**Review by larsnaesbye — COMMENTED** [review]
Looks good!

**Review by kenyon — APPROVED** [review]

File: README.md
Comment (alexjfisher) [inline README.md:118]: This terminology has been updated recently.  See https://puppet.com/docs/puppetdb/latest/release_notes.html#puppetdb-6130

I think the name of the file containing the allowlist has always been configurable, and it's just the configuration option that specifies this file that has been updated.  See https://puppet.com/docs/puppetdb/latest/configure.html#certificate-allowlist

@gdubicki Fantastic work on reworking this README, and I realise that this text is not new, just moved, but would you be able to update this section?

**Review by alexjfisher — CHANGES_REQUESTED** [review]

File: README.md
Comment (gdubicki) [reply README.md]: Sure @alexjfisher , I haven't noticed that. I replaced this whole fragment as this is not really a Puppet Enterprise-specific feature, at least as of now.

Comment (gdubicki) [reply README.md]: Oh, and thank you, @alexjfisher . 😅

**Comment by gdubicki** [comment]
@alexjfisher : can I get an approval please? If we find any issues in the README after the merge I will be happy to work on it more until it’s perfect. I really like this app and would like it to get more attention. :)

**Comment by gdubicki** [comment]
Please ping me if you need anything else from me to get this merged. Thanks!

File: requirements-test.txt
Comment (kenyon) [inline requirements-test.txt:5]: @gdubicki why this change?

Comment (gdubicki) [reply requirements-test.txt:5]: As I have written in the commit message of this change: because our tests are incompatible with Pytest 6. Perhaps I should have elaborated on that more, let me do that here. :)

I have squashed my commits so you won't find the link to a failed test in this PR, but after a while of digging in travis-ci.org/github/voxpupuli/puppetboard I found it here: https://travis-ci.org/github/voxpupuli/puppetboard/builds/755852596

In the output we have:
```
________________________ ERROR collecting test session _________________________

Direct construction of Pep8Item has been deprecated, please use Pep8Item.from_parent.

See https://docs.pytest.org/en/stable/deprecations.html#node-construction-changed-to-node-from-parent for more details.
```
...which, as explained under that URL, is something that:
```
[has] Changed in version 6.0. [of Pytest]
```

I would prefer to not update the tests for Pytest 6 in this PR. Firstly it is out of its scope. Secondly I am afraid that starting to work on in here would delay merging this PR to a point where it will start to be out of date or have to resolve conflicts that will arise.

**Pull request merged**
